src/engine/transforms/matchers.js

```diff
@@ -41,54 +43,59 @@ function transform(matchers: Array<Matcher>): Array<ParsedMatcher> {
       userDefinedSegmentMatcherData: segmentObject  /* segmentObject */,
       whitelistMatcherData: whitelistObject         /* whiteListObject */,
       unaryNumericMatcherData: unaryNumericObject   /* unaryNumericObject */,
-      betweenMatcherData: betweenObject             /* betweenObject */,
-      unaryStringMatcherData: unaryStringObject     /* unaryStringObject */
+      betweenMatcherData: betweenObject             /* betweenObject */
     } = matcher;
 
     let attribute = keySelector && keySelector.attribute;
     let type = matcherTypes.mapper(matcherType);
     let value = undefined;
+    let inputProcessor = undefined;
 
     if (type === matcherTypes.enum.SEGMENT) {
       value = segmentTransform(segmentObject);
     } else if (type === matcherTypes.enum.WHITELIST) {
       value = whitelistTransform(whitelistObject);
     } else if (type === matcherTypes.enum.EQUAL_TO) {
-      value = unaryNumericObject;
+      value = numericTransform(unaryNumericObject);
 
       if (unaryNumericObject.dataType === 'DATETIME') {
-        unaryNumericObject.value = zeroSinceHH(unaryNumericObject.value);
+        value = zeroSinceHH(value);
+        inputProcessor = zeroSinceHH;
```

Comment:
We should infer this just when needed.

Let discuss how we could move this into the `function sanitizeValue(...)`